Portal/vip-core/src/main/java/fr/insalyon/creatis/vip/core/server/dao/mysql/PlatformConnection.java

```diff
@@ -59,154 +59,164 @@
-            
-            createTable("VIPTermsOfuse",
-                    "idTermsOfuse INT(11) NOT NULL AUTO_INCREMENT, "
-                    + "date TIMESTAMP NULL, "
-                    + "text TEXT NULL, "
-                    + "PRIMARY KEY (idTermsOfuse)");
-
-            firstExecution = false;
+                createTable(connection,
+                            "VIPUsersGroups",
+                            "email VARCHAR(255), "
+                            + "groupname VARCHAR(100), "
+                            + "role VARCHAR(30), "
+                            + "PRIMARY KEY (email, groupname), "
+                            + "FOREIGN KEY (email) REFERENCES VIPUsers(email) "
+                            + "ON DELETE CASCADE ON UPDATE CASCADE, "
+                            + "FOREIGN KEY (groupname) REFERENCES VIPGroups(groupname) "
+                            + "ON DELETE CASCADE ON UPDATE CASCADE");
+
+                createTable(connection,
+                            "VIPAccounts",
+                            "name VARCHAR(255), "
+                            + "PRIMARY KEY (name)");
+
+                createTable(connection,
+                            "VIPAccountsGroups",
+                            "name VARCHAR(255), "
+                            + "groupname VARCHAR(255), "
+                            + "PRIMARY KEY (name, groupname), "
+                            + "FOREIGN KEY (name) REFERENCES VIPAccounts(name) "
+                            + "ON DELETE CASCADE ON UPDATE CASCADE, "
+                            + "FOREIGN KEY (groupname) REFERENCES VIPGroups(groupname) "
+                            + "ON DELETE CASCADE ON UPDATE CASCADE");
+
+                createTable(connection,
+                            "VIPPublication",
+                            "id INT(11) NOT NULL AUTO_INCREMENT, "
+                            + "title VARCHAR(255) NULL, "
+                            + "date VARCHAR(45) NULL, "
+                            + "doi VARCHAR(255) NULL, "
+                            + "autors VARCHAR(255) NULL, "
+                            + "type VARCHAR(255) NULL, "
+                            + "typeName VARCHAR(255) NULL, "
+                            + "VIPAuthor VARCHAR(255) NULL, "
+                            + "PRIMARY KEY (id), "
+                            + "FOREIGN KEY (VIPAuthor) REFERENCES VIPUsers(email) "
+                            + "ON DELETE CASCADE ON UPDATE CASCADE");
+
+                createTable(connection,
+                            "VIPTermsOfuse",
+                            "idTermsOfuse INT(11) NOT NULL AUTO_INCREMENT, "
+                            + "date TIMESTAMP NULL, "
+                            + "text TEXT NULL, "
+                            + "PRIMARY KEY (idTermsOfuse)");
+
+                firstExecution = false;
+            } catch (SQLException e) {
+                logger.error("Error closing database connection", e);
```

Comment:
How can we be sure that the SQLException is only concerned with closing the connection? 
In case it is concerned with something else, is it always enough to catch and log it ?

Reply:
I am sure of it.  The only code change was to add the fetching of the connexion.  The exception was not needed before that, so the sole case were it could happen is when closing the connection.

The strategy used in the rest of this function is to catch and log exception, see what is done with the DAOException.  So I implemented the same strategy.  This code is used when the servlet is initialised.  I am not sure what can be done.  Throw a RuntimeException ?

Reply:
If the only possible exception concerns the connection closing, then I would say that catching and logging it should be fine. But what if getConnection fails to retrieve the connection, isn't that possible ?

Reply:
You are right : the SQLException could also happen when getConnection is called.  I can change the message to : "Error while getting the connection or closing it".  Is that enough ?

Reply:
OK, go for the more explicit error message. I have no better suggestion.